### PR TITLE
Added MonadError instance to EitherT

### PR DESCRIPTION
**Pull request by 7h3kk1d**

### CODE REVIEW

File: src/main/java/com/jnape/palatable/lambda/monad/transformer/builtin/EitherT.java [hunk 1]
```diff
@@ -219,5 +238,4 @@ public String toString() {
             }
         };
     }
-
```
Comment (7h3kk1d) [inline src/main/java/com/jnape/palatable/lambda/monad/transformer/builtin/EitherT.java]: Accidental formatting. Will remove.

**Comment by jnape** [comment]
Sadly, I think this is going to screw `EitherT` in the future once we have a mechanism for transitively preserving properties from the base monad, as it (reasonably) appears to be an established convention to throw/catch into the base monad (see, for example, https://hackage.haskell.org/package/either-4.4.1/docs/Control-Monad-Trans-Either.html).

I think rather than incorporate `MonadError` in this way and then having to change it, maybe it's not worth going down this route with `EitherT`.

Does that seem fair?

**Comment by 7h3kk1d** [comment]
To recap our offline conversation. I think we were talking past each other a bit with regards to whether we were implementing `MonadError` or `MonadThrow`+`MonadCatch` specified in the above hackage link. It appears that `MonadThrow`+`MonadCatch` throws/catches with regards to the base `Monad` and `MonadError` is on the `EitherT` itself. There's some pros/cons of both approach and it's not clear if one is more obviously correct or intuitive in all scenarios over the other.

I'm cool with either approach or just delaying a decision till we have more information.

**Comment by jnape** [comment]
I've come around to this. I'll merge it in and release it with 5.2

**Comment by jnape** [comment]
Merged in d925e03.

**Pull request closed without merging**